api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDao.java

```diff
@@ -443,6 +453,49 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		return Optional.empty();
 	}
 	
+	/**
+	 * A handler for a {@link HasParam}
+	 *
+	 * @param propertyName the name of the property in the query to use
+	 * @param hasParam the {@link HasParam} to handle
+	 * @return a {@link Criterion} to be added to the query for the indicate has param
+	 */
+	protected Optional<Criterion> handleHas(String propertyName, HasParam hasParam) {
```

Comment:
Since we're *only* defining `_has` for `ServiceRequest`, this function should just be in the `FhirServiceRequestDaoImpl` class.